packages/react-aria-components/src/DropZone.tsx

```diff
@@ -41,38 +42,49 @@ export interface DropZoneRenderProps {
    * Whether the dropzone is the drop target.
    * @selector [data-drop-target]
    */
-  isDropTarget: boolean
+  isDropTarget: boolean,
+  /**
+   * Whether the button is currently in a pressed state.
+   * @selector [data-pressed]
+   */
+  isPressed: boolean
 }
 // note: possibly add isDisabled prop in the future
-export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps {}
+export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps, AriaLabelingProps, PressEvents {}
 
 export const DropZoneContext = createContext<ContextValue<DropZoneProps, HTMLDivElement>>(null);
 
 function DropZone(props: DropZoneProps, ref: ForwardedRef<HTMLDivElement>) {
   [props, ref] = useContextProps(props, ref, DropZoneContext);
+  let {onPress, onPressChange, onPressEnd, onPressStart, onPressUp} = props;
   let {dropProps, isDropTarget} = useDrop({...props, ref});
   let {hoverProps, isHovered} = useHover({});
   let {focusProps, isFocused, isFocusVisible} = useFocusRing({within: true});
 
   let buttonRef = useRef<HTMLButtonElement>(null);
   let inputRef = useRef<HTMLInputElement>(null);
-  let uploadLinkRef = useRef<HTMLAnchorElement>(null);
-  let uploadButtonRef = useRef<HTMLButtonElement>(null);
   let hasButton = useHasChild('button[slot=file]', ref);
   let hasLink = useHasChild('span[slot=file]', ref);
 
+  let isMobile = useIsMobileDevice();
```

Comment:
that's a good point
thinking about this more though, an iPad could have a keyboard attached to it, so I think we actually want to check current modality

then if it's keyboard modality, we say "press enter to select a file", vs if it's touch, we say "double tap..."
though is double tap for touch screen readers? i think users will know what to do if we just label it as "File system picker" or something along those lines. A VoiceOver user knows that they need to double tap because it'll announce as a button

Reply:
https://github.com/adobe/react-spectrum/blob/5a24e8147db1f596a0b047ffd88570ef935f5c29/packages/%40react-aria/interactions/src/useFocusVisible.ts#L181

Reply:
ah, right iPad + keyboard, good catch. I think labeling the button with "Open File system" can be sufficient, we can test it out. If it isn't, I'd recommend doing something like https://github.com/adobe/react-spectrum/blob/5a24e8147db1f596a0b047ffd88570ef935f5c29/packages/%40react-aria/table/src/useTableColumnResize.ts#L173-L176 and then checking if modality is "touch" for the mobile screen reader specific announcement

Reply:
so to clarify, we don't really need to check how the user is interacting with the dropzone. instead, we can just change the message to be more general and we assume that users will know how to interact with it.

Reply:
yep